catroid/src/org/catrobat/catroid/ui/fragment/BackPackSpriteFragment.java

```diff
@@ -218,14 +220,14 @@ public boolean onContextItemSelected(MenuItem item) {
 	}
 
 	private void checkForBackgroundUnpacking(Sprite selectedSprite, boolean delete, boolean keepCurrentSprite,
-			boolean fromHiddenBackPack) {
-		if (selectedSprite.isBackgroundObject) {
+			boolean fromHiddenBackPack, boolean asBackground) {
```

Comment:
Effective Java Item 25: "Long sequences of identically typed parameters are especially harmful."
```checkForBackgroundUnpacking(selectedSpriteBackPack, false, false, false, true);```

In the book there are two techniques explained how to improve it... but those may not be suited. Maybe you can improve it somehow. If not, no problem. It just caught my :eyes:...